modules/core/client/services/menus.client.service.js

```diff
@@ -155,9 +159,11 @@ angular.module('core').service('Menus', [
 
       // Search for menu item to remove
       for (var itemIndex in this.menus[menuId].items) {
-        for (var subitemIndex in this.menus[menuId].items[itemIndex].items) {
-          if (this.menus[menuId].items[itemIndex].items[subitemIndex].state === submenuItemState) {
-            this.menus[menuId].items[itemIndex].items.splice(subitemIndex, 1);
+        if (this.menus[menuId].items.hasOwnProperty(itemIndex)) {
+          for (var subitemIndex in this.menus[menuId].items[itemIndex].items) {
```

Comment:
Do you mean just one check of `hasOwnProperty`? We have to do it in both for loops, if we want to be sure it's the object's own property.